manifest/openapi/schema.go

```diff
@@ -96,6 +91,17 @@ func getTypeFromSchema(elem *openapi3.Schema, stackdepth uint64, typeCache *sync
 		return tftypes.Number, nil
 
 	case "":
+		if xv, ok := elem.Extensions["x-kubernetes-int-or-string"]; ok {
+			xb, err := xv.(json.RawMessage).MarshalJSON()
+			if err == nil {
```

Comment:
unfortunately `break` isn't a good fit here because it skips all the rest of the case block, while we still need to return DynamicPseudoType if all else fails. But I was able to reduce one nesting level by duplicating the final return statement instead of break.